lasagne/updates.py

```diff
@@ -113,3 +139,116 @@ def adadelta(loss, all_params, learning_rate=1.0, rho=0.95, epsilon=1e-6):
 
     return updates
 
+
+def norm_constraint(orig_update, param=None, abs_max=None, rel_max=None,
+                    epsilon=1e-7):
+    '''
+    Max weight norm constraints
+
+    This takes a parameter update and rescales it so that incoming weight
+    norms are below a specified constraint value.  The constraint value is
+    the lesser of `abs_max` (if provided) and `rel_max` (if provided) times
+    the average norm of the values originally stored in `param`.
+
+    Weight vectors violating the constraint are rescaled so that they are
+    within the allowed range.
+
+
+    :parameters:
+        - orig_update : TensorVariable
+            Theano expression for original update
+        - param : TheanoSharedVariable
+            Shared variable containing initial parameter values.
+            (Required if `rel_max` is used)
+        - abs_max : scalar
+            This value sets the absolute maximum value allowed for the
+            incoming weight vectors after the update.
+            (Optional)
```

Comment:
Minor note: If you write `(Optional)` here, you should do so for `rel_max` as well.